plugins/field-grid-dropdown/src/grid_item.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * Class representing an item in a grid.
+ */
+export class GridItem {
+  /** The DOM element for the grid item. */
+  private element: HTMLButtonElement;
+
+  /** Identifier for a click handler to unregister during dispose(). */
+  private clickHandler: Blockly.browserEvents.Data | null;
+
+  /** Callback to invoke when this item is selected. */
+  private selectionCallback?: (selectedItem: GridItem) => void;
+
+  /** Whether or not this item is currently selected. */
+  private selected = false;
+
+  /**
+   * Creates a new GridItem.
+   *
+   * @param container The parent element of this grid item in the DOM.
+   * @param content The content to display in this grid item.
+   * @param value The programmatic value of this grid item.
+   * @param selectionCallback Function to call when this item is selected.
+   */
+  constructor(
+    container: HTMLElement,
+    content: string | HTMLElement,
+    private readonly value: string,
+    selectionCallback: (selectedItem: GridItem) => void,
+  ) {
+    this.selectionCallback = selectionCallback;
+
+    this.element = document.createElement('button');
+    this.element.id = Blockly.utils.idGenerator.getNextUniqueId();
+    this.element.className = 'blocklyGridItem';
+    this.clickHandler = Blockly.browserEvents.conditionalBind(
+      this.element,
+      'click',
+      this,
+      this.onClick,
+      true,
+    );
+    container.appendChild(this.element);
+
+    const contentDom =
+      typeof content === 'string' ? document.createTextNode(content) : content;
+    this.element.appendChild(contentDom);
+
+    Blockly.utils.aria.setRole(this.element, Blockly.utils.aria.Role.GRIDCELL);
+  }
+
+  /**
+   * Disposes of this grid item.
+   */
+  dispose() {
+    this.selectionCallback = undefined;
```

Comment:
Why did you use undefined instead of null for the selection callback?

Reply:
For the nice ?: syntax, but moved to null for consistency.